gpkit/small_classes.py

```diff
@@ -93,6 +93,19 @@ def to_united_array(self, unitless_keys=(), united=False):
         "Converts all lists into array, potentially grabbing units from keys."
         _enray_and_unit_dict(self, self, unitless_keys, united)
 
+    def classify(self, cls):
+        _classify(self, cls)
+
+
+def _classify(d_in, cls):
+    "Converts all contained dictionaries into the given class."
+    for k, v in d_in.items():
+        if isinstance(v, dict) and v:
+            if isinstance(v.keys()[0], Strings):
+                _classify(v, cls)
+            else:
+                d_in[k] = cls(v)
```

Comment:
I don't understand this implementation. It doesn't seem to do what the docstring says. What is the significance of Strings? Here is an example where I expected the subdict to get converted to an A but it didn't:

```python
class A(dict):
    def __repr__(self):
        return "A"

d = {5: {6: 7, "hi": 8}, "hi": 9}
_classify(d, A)
print d   # {'hi': 9, 5: {'hi': 8, 6: 7}}
```